externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala

```diff
@@ -53,7 +54,13 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
     val sessionImpl = new SparkSessionImpl(protocol, user, password, ipAddress, conf, this)
     val handle = sessionImpl.handle
     try {
-      val sparkSession = spark.newSession()
+      val sparkSession = if (spark.sessionState.conf
+        .getConf(StaticSQLConf.HIVE_THRIFT_SERVER_SINGLESESSION)) {
```

Comment:
Can we add a new config in `KyuubiConf` ? We don't want to introduce any thriftserver config.

Reply:
And don't forget to generate the docs with cmd
`KYUUBI_UPDATE=1 build/mvn test -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration`

Reply:
> Can we add a new config in `KyuubiConf` ? We don't want to introduce any thriftserver config.

Is it okay to use `kyuubi.session.engine.singleSession` configuration name?

Reply:
We can make it more simple `kyuubi.engine.singleton`

Reply:
FYI @turboFei @pan3793 @yaooqinn

Reply:
kyuubi.engine.singleton looks overkill

Reply:
how about `kyuubi.engine.single.spark.session`? `kyuubi.engine.singleton` seems to tell users that only one engine will be created

Reply:
+1 for `kyuubi.engine.single.spark.session`

Reply:
`kyuubi.engine.single.spark.session` looks good